docs/docs/resources/pipeline-components/dependencies/kpops_structure.yaml

```diff
@@ -85,13 +85,76 @@ kpops_components_fields:
   - repo_config
   - version
 kpops_components_inheritance_ref:
-  helm-app: kubernetes-app
-  kafka-app: pipeline-component
-  kafka-connector: pipeline-component
-  kafka-sink-connector: kafka-connector
-  kafka-source-connector: kafka-connector
-  kubernetes-app: pipeline-component
-  pipeline-component: base-defaults-component
-  producer-app: kafka-app
-  streams-app: kafka-app
-  streams-bootstrap: helm-app
+  helm-app:
+    bases:
+    - kubernetes-app
+    parents:
+    - helm-app
+    - kubernetes-app
+    - pipeline-component
+  kafka-app:
+    bases:
+    - pipeline-component
+    parents:
+    - kafka-app
+    - pipeline-component
+  kafka-connector:
+    bases:
+    - pipeline-component
+    parents:
+    - kafka-connector
+    - pipeline-component
+  kafka-sink-connector:
+    bases:
+    - kafka-connector
+    parents:
+    - kafka-sink-connector
+    - kafka-connector
+    - pipeline-component
+  kafka-source-connector:
+    bases:
+    - kafka-connector
+    parents:
+    - kafka-source-connector
+    - kafka-connector
+    - pipeline-component
+  kubernetes-app:
+    bases:
+    - pipeline-component
+    parents:
+    - kubernetes-app
+    - pipeline-component
+  pipeline-component:
+    bases: []
+    parents:
+    - pipeline-component
+  producer-app:
+    bases:
+    - kafka-app
+    - streams-bootstrap
+    parents:
+    - producer-app
+    - kafka-app
+    - streams-bootstrap
+    - helm-app
+    - kubernetes-app
+    - pipeline-component
+  streams-app:
+    bases:
+    - kafka-app
+    - streams-bootstrap
+    parents:
+    - streams-app
```

Comment:
is it correct that the class itself is listed as parent?

Reply:
It is intended, yes.